### DISCUSSION TITLE
Collaboration on additional features

### DISCUSSION THREAD

**Discussion by austinrdennis** [ColormaticStudios/quality-godot-first-person-2#30] [General]
@CZakarya First off, thanks a lot for making this and putting in the AssetLib. You're GOATed for that!

Bottom line up front (BLUF): Are you willing/able to take on some targeted collaboration on this? 

I'd like to expand some of what you got since I think some stuff I implemented in my game would be useful in most other games too. If you aren't available or don't desire that, I totally get it and read no further. Just tell me no in the replies and off I go.

If you're interested, what's the scope of this project? Also, what kind of time commitment are you willing to give to this project from an admin perspective? I'd love to tweak/add some features but I'm not trying to load you up with a bunch of pull requests you'll never have time for or introduce an unmanageable amount of features/systems.

I'm not a wiz kid for GDScript or Godot in general, but I've been making the stuff I added work the way I want without much fuss. It could use a refactor and clean up, but it's generally sound I think. Let me know what big ticket things you're looking to add or refine and I'll take a crack at it.

**Comment by CZakarya** [comment]
I would say to start, just fork this project and make changes to that and I will check them out. I can merge or re-implement the ones I think would be good to have.

Over the course of this project, I've had ideas for more systems, like an item pickup/interaction system. Originally, there was going to be a third-person sister project to this but I never got around to that. I think if I want to start a new project related to this I would make a fully modular 3D game starter.

I would definitely like to have someone else contribute their ideas. One thing you could work on is the editor module, I haven't really completed it's implementation, and it needs some love. It's purpose is to modify properties of the FPC in the editor, and have those modifications apply during runtime.

**Reply by austinrdennis** [reply]
I hear ya, I'll polish up the editor exports when I have some spare time. I'll ping you when it's worth a look and we can go from there. The game starter sounds like a fun project and Godot desperately needs that kind of thing in general.

**Comment by austinrdennis** [comment]
@CZakarya What's your intended use case for the EditorModule node? Based on the code it's not doing anything at runtime currently, but you've got plans for it.
